tests.py

```diff
@@ -17,30 +17,19 @@
 matplotlib.use('agg')
 
 import nose
-from nose.plugins import attrib
-from matplotlib.testing.noseclasses import KnownFailure
 from matplotlib import default_test_modules
 
-plugins = [KnownFailure, attrib.Plugin]
-
-# Nose doesn't automatically instantiate all of the plugins in the
-# child processes, so we have to provide the multiprocess plugin with
-# a list.
-from nose.plugins import multiprocess
-multiprocess._instantiate_plugins = plugins
-
 
 def run(extra_args):
-    try:
-        import faulthandler
-    except ImportError:
-        pass
-    else:
-        faulthandler.enable()
+    from nose.plugins import multiprocess
+
+    matplotlib._init_tests()
 
-    if not os.path.isdir(
```

Comment:
Re: mock: Since it's included in Python 3.3 and onwards, maybe I'll update this to install it on Python 2.7 only (which shouldn't require the weirdness we had for 2.6, which I'll just drop since 2.6 is going away anyway).